pulsar-broker/src/test/java/org/apache/pulsar/common/naming/SpecifiedPositionsBundleSplitAlgorithmTest.java

```diff
@@ -0,0 +1,87 @@
+
+import com.google.common.collect.Lists;
+import java.util.Arrays;
+import org.apache.pulsar.broker.namespace.NamespaceService;
+import org.testng.annotations.Test;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class SpecifiedPositionsBundleSplitAlgorithmTest {
+    @Test
+    public void testNullBoundaries() throws Exception {
+        SpecifiedPositionsBundleSplitAlgorithm algorithm = new SpecifiedPositionsBundleSplitAlgorithm();
+        assertNull(algorithm.getSplitBoundary(new BundleSplitOption()).get());
+    }
+
+    @Test
+    public void testTotalTopicsSizeLessThan1() {
+        SpecifiedPositionsBundleSplitAlgorithm algorithm = new SpecifiedPositionsBundleSplitAlgorithm();
+        NamespaceService mockNamespaceService = mock(NamespaceService.class);
+        NamespaceBundle mockNamespaceBundle = mock(NamespaceBundle.class);
+        doReturn(CompletableFuture.completedFuture(Lists.newArrayList("a")))
+                .when(mockNamespaceService).getOwnedTopicListForNamespaceBundle(mockNamespaceBundle);
+        assertNull(algorithm.getSplitBoundary(new BundleSplitOption(mockNamespaceService, mockNamespaceBundle,
+                Arrays.asList(1L, 2L))).join());
+    }
+
+    @Test
+    public void testSpecifiedPositionsLessThan1() {
+        SpecifiedPositionsBundleSplitAlgorithm algorithm = new SpecifiedPositionsBundleSplitAlgorithm();
+        NamespaceService mockNamespaceService = mock(NamespaceService.class);
+        NamespaceBundle mockNamespaceBundle = mock(NamespaceBundle.class);
+        assertNull(algorithm.getSplitBoundary(
+                new BundleSplitOption(mockNamespaceService, mockNamespaceBundle, null)).join());
+        assertNull(algorithm.getSplitBoundary(
+                new BundleSplitOption(mockNamespaceService, mockNamespaceBundle, new ArrayList<>())).join());
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    @Test
+    public void testAlgorithmReturnCorrectResult() {
+        // -- algorithm
+        SpecifiedPositionsBundleSplitAlgorithm algorithm = new SpecifiedPositionsBundleSplitAlgorithm();
+        // -- calculate the mock result
+        NamespaceService mockNamespaceService = mock(NamespaceService.class);
+        NamespaceBundle mockNamespaceBundle = mock(NamespaceBundle.class);
+        doReturn(1L).when(mockNamespaceBundle).getLowerEndpoint();
+        doReturn(1000L).when(mockNamespaceBundle).getUpperEndpoint();
+        doReturn(CompletableFuture.completedFuture(Lists.newArrayList("topic", "topic2")))
+                .when(mockNamespaceService)
+                .getOwnedTopicListForNamespaceBundle(mockNamespaceBundle);
+
+        List<Long> positions = Arrays.asList(100L, 200L, 500L, 800L);
```

Comment:
Please add some more corner case, like negative value, LowerEndpoint, UpperEndpoint, out of range value, etc.

Reply:
sure